### PR TITLE
[1LP][RFR] Fix navigation for server

### PR DESCRIPTION
**Pull request by mshriver**
create tree_path property for Server
Use it in navigation/views where this tree path is built from server attributes
add base session fixture to set appliance name

There are some other failures going on in PRT, but I don't really want this PR scope creeping too far beyond the changes here related to server configuration navigation/view assertions.

{{ pytest: cfme/tests/integration/test_cfme_auth.py::test_login_evm_group --long-running }}

### CODE REVIEW

File: cfme/fixtures/base.py [hunk 1]
```diff
@@ -58,3 +58,19 @@ def fix_missing_hostname(appliance):
 
             if ssh_client.run_command('grep $(hostname) /etc/hosts').failed:
                 logger.error('Failed to mangle /etc/hosts')
+
+
+@pytest.fixture(scope="session", autouse=True)
```
Comment (mshriver) [inline cfme/fixtures/base.py]: @izapolsk gonna hate me for this :)

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 0c75f758e2ad6d0cf87ae5c9c159df56b9c41b80

 The *global* fixture **`fix_missing_appliance_name`** was changed, but I didn't find where it's used.
Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/base/__init__.py [hunk 2]
```diff
@@ -65,25 +85,37 @@ def collect_logs(self):
 
     @property
     def zone(self):
-        server_res = self.appliance.rest_api.collections.servers.find_by(id=self.sid)
-        server, = server_res
+        server, = self.appliance.rest_api.collections.servers.find_by(id=self.sid)
```
Comment (john-dupuy) [inline cfme/base/__init__.py]: Q: can you use `self.rest_api_entity` here?

File: cfme/fixtures/base.py [hunk 3]
```diff
@@ -58,3 +58,19 @@ def fix_missing_hostname(appliance):
...
+def fix_missing_appliance_name(appliance):
+    """Fix for an empty appliance server name"""
+    if isinstance(appliance, DummyAppliance) or appliance.is_dev:
+        return
+    logger.info('Checking appliance server name is set')
+    if appliance.server.name == '':
```
Comment (john-dupuy) [inline cfme/fixtures/base.py]: Q: Why can't you just do?

```python,
if not appliance.server.name:
```

File: cfme/infrastructure/host.py [hunk 4]
```diff
@@ -457,7 +458,10 @@ def get_credentials_from_config(cls, credentials_config):
 
     @property
     def rest_api_entity(self):
-        return self.appliance.rest_api.collections.hosts.get(name=self.name)
+        try:
+            return self.appliance.rest_api.collections.hosts.get(name=self.name)
+        except ValueError:
+            raise RestLookupError(f'No host rest entity found matching name {self.name}')
```
Comment (john-dupuy) [inline cfme/infrastructure/host.py]: :tada:

File: cfme/base/__init__.py [hunk 5]
```diff
@@ -46,6 +47,25 @@ def name(self):
...
+    def tree_path(self):
+        """Generate the path list for navigation purposes
+        list elements follow the tree path in the configuration accordion
+
+        Returns:
+            list of path elements for tree navigation
+        """
+        current = '' if self.is_slave else ' (current)'
+        # replace two spaces with one, formatting tweak when name is empty
+        name_string = f' {self.name} '.replace('  ', ' ')
```
Comment (kedark3) [inline cfme/base/__init__.py]: Can you tell me more about what are you doing here ? Not sure if I am familiar with what this property does and what's up with the spaces?

File: cfme/base/__init__.py (hunk 2 above)
Comment (kedark3) [inline cfme/base/__init__.py]: why do we have `,` after `server` ? wouldn't it throw ValueError if there are multiple values?

File: cfme/exceptions.py [hunk 6]
```diff
@@ -277,6 +277,10 @@ class NeedleNotFoundInLog(CFMEException):
     """Raised when log doesnt't contain needle"""
 
 
+class RestLookupError(CFMEException):
+    """raised when lookup of a rest entity fails"""
+
+
```
Comment (kedark3) [inline cfme/exceptions.py]: Do we not want to define any specific error message here?

File: cfme/base/__init__.py [hunk 7]
```diff
@@ -65,25 +85,37 @@ def collect_logs(self):
...
         return '/'.join(
             [self.appliance.rest_api.collections.servers._href, str(self.sid), 'settings']
         )
 
+    @property
+    def rest_api_entity(self):
+        try:
+            return self.appliance.rest_api.collections.servers.get(id=self.sid)
+        except ValueError:
+            raise RestLookupError
```
Comment (kedark3) [inline cfme/base/__init__.py]: No message here? Like you have at https://github.com/ManageIQ/integration_tests/pull/9382/files#diff-a7dcf56bdad11918a6877cbbf1a10059R158

File: cfme/fixtures/base.py (hunk 3 above)
Comment (mshriver) [reply cfme/fixtures/base.py]: can, just being as explicit as possible.

File: cfme/base/__init__.py (hunk 5 above)
Comment (mshriver) [reply cfme/base/__init__.py]: When the name attribute is an empty string, you end up with two spaces. Trying to format this into the complete string in which its used, there has to be leading/following spaces, but only one space when the name is empty.  This just injects the name, and then collapses two spaces to one if the name was empty - without checking what self.name is and having a ternary.

File: cfme/base/__init__.py (hunk 2 above)
Comment (mshriver) [reply cfme/base/__init__.py]: yes, which is the current behavior, and intended. If you have mutliple server entities with the same ID, that's a problem.

File: cfme/exceptions.py (hunk 6 above)
Comment (mshriver) [reply cfme/exceptions.py]: Not really, whatever exception message occurred in the underlying exception class would carry through.

File: cfme/base/__init__.py (hunk 7 above)
Comment (mshriver) [reply cfme/base/__init__.py]: I can add one here as well.

File: cfme/base/__init__.py (hunk 2 above)
Comment (kedark3) [reply cfme/base/__init__.py]: Can you comment it there maybe?

Comment (mshriver) [reply cfme/base/__init__.py]: Will check it out

File: cfme/base/__init__.py [hunk 8]
```diff
@@ -114,7 +112,7 @@ def rest_api_entity(self):
         try:
             return self.appliance.rest_api.collections.servers.get(id=self.sid)
         except ValueError:
-            raise RestLookupError
+            raise RestLookupError(f'No server rest entity found matching ID {self.sid}')
```
Comment (kedark3) [inline cfme/base/__init__.py]: :tada:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 493a2b31b684db54e9ec22fcc01496203430e435

 The *global* fixture **`fix_missing_appliance_name`** was changed, but I didn't find where it's used.
The *local* fixture **`user_obj`** is used in the following files:
- **cfme/tests/integration/test_cfme_auth.py**
    - *test_login_evm_group*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 2f609282f44c1bf947b9ecdc4936b4faf8a64a80

 The *global* fixture **`fix_missing_appliance_name`** was changed, but I didn't find where it's used.
The *local* fixture **`user_obj`** is used in the following files:
- **cfme/tests/integration/test_cfme_auth.py**
    - *test_login_evm_group*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/integration/test_cfme_auth.py [hunk 9]
```diff
@@ -134,7 +135,7 @@ def test_login_evm_group(appliance, auth_user, user_obj, soft_assert):
     evm_group_names = [group for group in auth_user.groups if 'evmgroup' in group.lower()]
     with user_obj:
         logger.info('Logging in as user %s, member of groups %s', user_obj, evm_group_names)
-        view = navigate_to(appliance.server, 'LoggedIn')
+        view = navigate_to(appliance.server, 'Dashboard')
```
Comment (digitronik) [inline cfme/tests/integration/test_cfme_auth.py]: :+1:

File: cfme/fixtures/base.py (hunk 1 above)
Comment (izapolsk) [reply cfme/fixtures/base.py]: no worries, I'll get rid of it soon :)

**Pull request merged**